### ISSUE TITLE
New xalienfs breaks compilation

### ISSUE THREAD

**Issue by dstocco** [alisw/alidist#512]
Dear all,
the commit: cb00b0a34172e384b27a55efb97ae2b6d6e5410d
breaks the compilation both on Mac OSX and SL6.

I'm attaching the log for SL6. Can it be reverted?

Thanks in advance,
best regards,
Diego

[log.txt](https://github.com/alisw/alidist/files/442142/log.txt)

**Comment by dberzano** [comment]
Hello Diego, I have fixed it far more than 14 minutes ago. The problem is a wrong tag in the xalienfs repo, and not that commit. Can you just try again?

For the record the only thing that changes is alisw/xalienfs#2.

**Comment by dstocco** [comment]
Hi Dario,
I'm trying to try again, but I cannot test it.

I went on alidist, on branch IB/v5-08/next
then did:
`git pull`
and then launched alibuild, but for some reason now I get:
```
Username for 'https://gitlab.cern.ch': dstocco
Password for 'https://dstocco@gitlab.cern.ch'
```
And it stops with this error:
`ERROR: Unable to fetch from https://gitlab.cern.ch/ALICEPrivateExternals/DPMJET.git`

Did I make some mistake?

Thanks,
cheers,
Diego

**Comment by dberzano** [comment]
Hello Diego, note that next == unstable. I have just rebased the next branch so you should have run `git fetch && git reset --hard origin/IB/v5-08/next` instead of a plain `git pull`. In practice this problem is unrelated. If you update now, it should go away.

**Comment by dstocco** [comment]
Hi Dario,
I figured it was unrelated, but I can't get rid of it.
I tried with:
`git fetch && git reset --hard origin/IB/v5-08/next`
but it still asks for the gitlab pwd.

Anyways, I also have an o2 version, with IB/v5-08/o2.
I'm not asked for a gitlab password for this.
So I tried to updated and see if the xalinefs issue is fixed there (on Mac OSX).
You can find the log in attachment.
Thans,
cheers,
Diego
[log_o2.txt](https://github.com/alisw/alidist/files/442314/log_o2.txt)

**Comment by dstocco** [comment]
Hi Dario,
small follow-up: I got rid of the gitlab message. Sorry for the noise.
On the other hand, I decided to stick to the IB/v5-08/prod, so that I should not have issues with the unstable version.

This said, the issue on IB/v5-08/o2 still holds (see previous comment).
Thanks in advance,
ciao,
Diego

**Comment by dberzano** [comment]
As soon as next is merged into prod (#514) we'll fix o2.

**Comment by dstocco** [comment]
Ciao Dario,
since you merged on IB/v5-08/prod, I tried again with the AliPhysics master, which is now broken both on Mac and SL6.
So it seems that the "stability" of the "next" branch was not the problem.

I attach both log files for debugging purposes.
Thanks in advance,
ciao,
Diego

[log_prod_mac.txt](https://github.com/alisw/alidist/files/444290/log_prod_mac.txt)
[log_prod_sl6.txt](https://github.com/alisw/alidist/files/444289/log_prod_sl6.txt)

**Comment by dberzano** [comment]
Hello Diego, it is possible that your local xalienfs mirror did not get the moved tag right (it looks it's the case from your error), so:

```
rm -rf MIRROR/xalienfs
rm -rf SOURCES/xalienfs
```

under your `sw` directory. For the record your local setup has issues but the upstream recipes are not broken.

**Comment by dstocco** [comment]
Ciao Dario,
ok I think I see what happened. When I first updated alidist (in branch next), just after the new xalienfs was added, the xalienfs was still buggy in the repository. So it was that one that was downloaded on my system.
You then fixed the xalienfs on the repository...but alibuild was using the one that it found in my system under MIRROR.

I removed the MIRROR and SOURCES and started again: it is working now.

I guess that, to avoid similar issues in the future, it's better if I always use the "prod" branch instead of the "next" one (had to change for testing purposes...forgot to come back).

For the records: this was also the issue for O2. It is working now. Closing the ticket.

Thanks a lot!
Ciao,
Diego
